### PR TITLE
chore: reintroduce no-shadow in minimal form

### PR DESCRIPTION
**Pull request by peter-sanderson**
Solves: https://github.com/trezor/trezor-suite/issues/15311

### CODE REVIEW

**Review by Lemonexe — APPROVED** [review]

File: packages/eslint/src/typescriptConfig.mjs [hunk 1]
```diff
@@ -5,22 +5,6 @@ export const typescriptConfig = [
...
-            '@typescript-eslint/no-unused-vars': [
-                'error',
-                {
-                    vars: 'all',
-                    args: 'none',
-                    ignoreRestSiblings: true,
-                    varsIgnorePattern: '^_',
-                },
-            ],
-            '@typescript-eslint/no-shadow': [
```
Comment (Nodonisko) [inline packages/eslint/src/typescriptConfig.mjs:17]: It was not off, it was on only for new packages from moment I added it. Please bring it back at least for suite-native

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: packages/eslint/src/typescriptConfig.mjs (hunk 1 above)
Comment (peter-sanderson) [reply packages/eslint/src/typescriptConfig.mjs:17]: I had some hard time with this rule: I created a follow-up issue here: https://github.com/trezor/trezor-suite/issues/15311

Comment (peter-sanderson) [reply packages/eslint/src/typescriptConfig.mjs:17]: Oh, I missed you comment there. I try for native.

Comment (peter-sanderson) [reply packages/eslint/src/typescriptConfig.mjs:17]: done

**Review by Nodonisko — APPROVED** [review]

**Review by Lemonexe — APPROVED** [review]
Looks.. Acceptable to me :neutral_face: 

Renaming `variable` to `variable2` is not pretty, but if the goal is to annoy us developers so that we rename it appropriately when working on these files in future, then I guess it's better than `disable-next-line`, as it will prevent bugs, and certainly give enough annoyance :laughing: 

:rocket:

**Pull request merged**